airflow-core/src/airflow/example_dags/example_dag_decorator.py

```diff
@@ -22,13 +22,21 @@
 
 import httpx
 import pendulum
-
+import ipaddress
 from airflow.providers.standard.operators.bash import BashOperator
 from airflow.sdk import BaseOperator, dag, task
 
 if TYPE_CHECKING:
     from airflow.sdk import Context
 
+def is_ip(address: str) -> bool:
+    """Check if the provided address is a valid IP address (supports IPv4 and IPv6)."""
+    try:
+        # Try to convert the address to an IP address (supports both IPv4 and IPv6)
+        ipaddress.ip_address(address)
+        return True
+    except ValueError:
+        return False
```

Comment:
Maybe try:
```
raise AirflowException(f"Invalid IP address: '{external_ip}'.")
```

Reply:
I do not see a real reason to do that, any exception should do tbh. This is mostly a one-off case and adding a new type solely for that doesn't justify it to me

Reply:
There was a discussion long ago (on Slack, so it's no longer there) about not using `AirflowException` when not needed, as it's vague.

If any exception should do, then maybe we could try 


```python
raise ValueError(f"Invalid IP address: '{external_ip}'.")
```

Reply:
Oh I do not recall that but it does make sense. ValueError works